beef/merkletree.go

```diff
@@ -0,0 +1,42 @@
+package beef
+
+import (
+	"encoding/hex"
+
+	crypto "github.com/bitcoin-sv/go-sdk/primitives/hash"
+	"github.com/bitcoin-sv/go-sdk/util"
+)
+
+// INFO: This function is moved to go-paymail from go-bt
+// try to use go-sdk implementation when available
+
+// merkleTreeParentStr returns the Merkle Tree parent of two Merkle
+func merkleTreeParentStr(leftNode, rightNode string) (string, error) {
+	l, err := hex.DecodeString(leftNode)
+	if err != nil {
+		return "", err
+	}
+	r, err := hex.DecodeString(rightNode)
+	if err != nil {
+		return "", err
+	}
+
+	return hex.EncodeToString(merkleTreeParent(l, r)), nil
+}
+
+// merkleTreeParent returns the Merkle Tree parent of two Merkle
+// Tree children.
+func merkleTreeParent(leftNode, rightNode []byte) []byte {
+	// swap endianness before concatenating
+	l := util.ReverseBytes(leftNode)
+	r := util.ReverseBytes(rightNode)
+
+	// concatenate leaves
+	concat := append(l, r...)
```

Comment:
To achieve this, you had to:
- reserve memory for L 
- reserve memory for R
- resize the L (during append) which is basically reserving memory for L+R and making a copy

I don't know if it's frequently used by spv-wallet, but to speed it up you could reserve only one array, like this:
https://go.dev/play/p/s-PeekV1TMM

Reply:
It is used only once in MerkleRootVerification - I didn't want to write or rewrite it because I hope it will be added in go-sdk (I will ask a request for that in the scope of SPV-1035